publiq-ui/pub-checkbox.vue

```diff
@@ -0,0 +1,29 @@
+<template>
+  <b-form-checkbox
+    :checked="isChecked"
+    :disabled="isDisabled"
+    @input="handleInput"
+  >
+    <slot />
+  </b-form-checkbox>
+</template>
+
+<script>
+  export default {
+    props: {
+      isChecked: {
+        type: Boolean,
+        default: false,
+      },
+      isDisabled: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    methods: {
+      handleInput() {
+        this.$emit('input');
```

Comment:
Is the name `input` a Vue convention, a Bootstrap convention, or another kind of convention for input changes? If it's a Bootstrap convention I'm not sure we should re-use it. The Bootstrap components should be the implementation we use underneath, but should not define the public API of our custom components. That way we can easily switch later if we do a redesign and maybe switch to e.g. Material Design components.

Reply:
Other question: shouldn't we pass some kind of event object to `this.$emit('input');` containing the state of the checkbox? Or how would a listener know that state?

Reply:
The way we do it right now is that the parent component knows which checkbox is checked and passes it on to each checkbox by the isChecked prop. The input handler is used to toggle the known boolean state of that checkbox. We emit this event because we keep the state of the checkbox not in the checkbox itself but in the parent components

Reply:
Okay thanks, that answers my second question. My first question still stands 😊

Reply:
I think the `input` handler is not necessarily a vue convention, in this case a bootstrap specific event I think. But what would be a better alternative then. To rename the emitted event to something more specific like `select`?

Reply:
`toggle` perhaps? Or `check`? Although the latter might indicate it's `checked` while it seems like the event is also emitted when unchecking. So that's why I'd go with `toggle` personally if you agree.

Reply:
done